### PR TITLE
v5.0.x: docs: add documentation on ROCm devices

### PR DESCRIPTION
**Pull request by edgargabriel**
Signed-off-by: Edgar Gabriel <Edgar.Gabriel@amd.com>
(cherry picked from commit 9bdaf3e77c97f2f98e255362be975bfe398772cc)

### CODE REVIEW

**Review by jsquyres — APPROVED** [review]

**Comment by awlauria** [comment]
@edgargabriel does this depend on #10911 ? If so I'll wait to merge together.

**Comment by edgargabriel** [comment]
technically, no. Practically, probably yes. I think it is fine to merge it after the accelerator framework has been merged. More important is probably the update to the authors file.

**Comment by jsquyres** [comment]
Now that https://github.com/open-mpi/ompi/pull/10911 has been merged, this PR is good to go.

**Pull request merged**
